### PR TITLE
Nefjekyll refactor

### PR DESCRIPTION
**Pull request by miguelangel-dev**
## Issues
Related issue #112

## Description
In this PR we have updated the `NefJekyll` module for using the new NefCore+NefRender FP. New `nef` APIs for IO and EnvIO

### CODE REVIEW

File: project/Component/NefJekyll/NefJekyll.swift [hunk 1]
```diff
@@ -10,102 +10,173 @@ import Bow
...
+    
+    // MARK: private <steps>
+    private func buildMainPage(_ mainPage: URL, docs: URL) -> EnvIO<Environment, RenderError, Void> {
+        let file = docs.appendingPathComponent("README.md")
+        let content = (try? String(contentsOf: mainPage)) ?? """
+                                                             ---
+                                                             layout: docs
+                                                             permalink: /docs/
+                                                             ---
+                                                             """
```
Comment (truizlop) [inline project/Component/NefJekyll/NefJekyll.swift]: This string hanging out in the middle of the code... I think it should be extracted to a constant, or at least make sure to align it to the left of the code.

File: project/Component/nef/JekyllAPI.swift [hunk 2]
```diff
@@ -32,43 +31,57 @@ public extension JekyllAPI {
...
+    static func renderVerboseIO(content: String, permalink: String, toFile file: URL) -> IO<nef.Error, (url: URL, ast: String, rendered: String)> {
+        renderVerbose(content: content, permalink: permalink, toFile: file).provide(MacDummyConsole())
+    }
+    
+    static func renderIO(playground: URL, into output: URL) -> IO<nef.Error, NEA<URL>> {
+        render(playground: playground, into: output).provide(MacDummyConsole())
+    }
+    
+    static func renderIO(playgroundsAt: URL, mainPage: URL, into output: URL) -> IO<nef.Error, NEA<URL>> {
+        render(playgroundsAt: playgroundsAt, mainPage: mainPage, into: output).provide(MacDummyConsole())
```
Comment (truizlop) [inline project/Component/nef/JekyllAPI.swift:76]: This should be in the protocol declaration as extension methods.

File: project/Component/nef/MarkdownAPI.swift [hunk 3]
```diff
@@ -50,12 +51,37 @@ public extension MarkdownAPI {
...
+    }
+    
+    static func renderIO(playground: URL, into output: URL) -> IO<nef.Error, NEA<URL>> {
+        render(playground: playground, into: output).provide(MacDummyConsole())
+    }
+    
+    static func renderIO(playgroundsAt folder: URL, into output: URL) -> IO<nef.Error, NEA<URL>> {
+        render(playgroundsAt: folder, into: output).provide(MacDummyConsole())
+    }
+
```
Comment (truizlop) [inline project/Component/nef/MarkdownAPI.swift:78]: Same here.

**Review by truizlop — APPROVED** [review]

File: project/Component/nef/JekyllAPI.swift (hunk 2 above)
Comment (miguelangel-dev) [reply project/Component/nef/JekyllAPI.swift:76]: offline-conversation: we only will use the `EnvIO` API and will add support to IO (provide dummy console in `unsafeRun` with some utils)

**Pull request merged**